app/components/UI/FiatOnRampAggregator/Views/AmountToBuy.js

```diff
@@ -666,11 +666,28 @@ const AmountToBuy = () => {
                 label={strings('fiat_on_ramp_aggregator.amount')}
                 currencySymbol={currentFiatCurrency?.denomSymbol}
                 amount={displayAmount}
+                highlightedError={
+                  amountNumber !== 0 &&
```

Comment:
It could be a nice optimization to memorize the three different conditions and check against in different parts of your UI rendering logic as following:
```suggestion
                  const amountIsValid = useMemo(()=>{
                  amountNumber > 0 && 
                  (amountNumber > limits.minAmount ||
                    amountNumber < limits.maxAmount)
                    }, [amountNumber, limits.minAmount, limits.maxAmount]);
                    
                  const amountIsBelowMinimum = useMemo(()=>{
                  amountNumber !== 0 && 
                  amountNumber < limits.minAmount
                  }, [amountNumber, limits.minAmount]);
                  
                  const amountIsAboveMaximum = useMemo(()=>{
                  amountNumber !== 0 && 
                  amountNumber > limits.maxAmount
                  }, [amountNumber, limits.maxAmount]);
```

Reply:
Good suggestion, added